launcher/MangoHud.cpp

```diff
@@ -111,10 +112,18 @@ QString getLibraryString()
         auto conf = Json::requireDocument(filePath, vkLayer);
         auto confObject = Json::requireObject(conf, vkLayer);
         auto layer = Json::ensureObject(confObject, "layer");
-        return Json::ensureString(layer, "library_path");
+        QString libraryName = Json::ensureString(layer, "library_path");
+
+        // Check whether mangohud is usable
+        if (!libraryName.isEmpty()) {
+            QString libraryPath = findLibrary(libraryName);
+            if (!libraryPath.isEmpty()) {
+                return libraryPath;
+            }
+        }
```

Comment:
Wouldn't it be best to just return the libraryName if the findLibrary fails to find it?
That way we are consistent with old behavior.
```suggestion
            }
            return libraryName;
        }
```

Reply:
`findLibrary()` also doubles as a check that this shared object indeed can be dlopen and loaded into a process.

Not sure what merit this would have to return a string when the library is not useable.

Reply:
maybe I misunderstood but `findLibrary` only works for glibc and not clang.
So in case of clang the dlopen will not happen and `findLibrary` will return an empty string.

My only requirement here is in case `findLibrary` fails in any way to return back the `libraryName` like we used to until now

Reply:
Ah I see your concerns now - dlopen would only work on linux based systems (afaik not related to specific compiler, whether that's gcc or clang. It will work as long as system C library supports `dlopen()`, whether that's glibc or musl or something).

Will push an amend shortly.